Riot/Modules/Authentication/AuthenticationViewController.m

```diff
@@ -869,4 +869,114 @@ - (void)authInputsViewDidCancelOperation:(MXKAuthInputsView *)authInputsView
     [self cancel];
 }
 
+- (void)authInputsView:(MXKAuthInputsView *)authInputsView autoDiscoverServerWithDomain:(NSString *)domain
+{
+    [self tryServerDiscoveryOnDomain:domain];
+}
+
+#pragma mark - Server discovery
+
+- (void)tryServerDiscoveryOnDomain:(NSString *)domain
+{
+    MXAutoDiscovery *autoDiscovery = [[MXAutoDiscovery alloc] initWithDomain:domain];
```

Comment:
MXAutoDiscovery instance should be retained by `AuthenticationViewController`.  Normally `MXAutoDiscovery *autoDiscovery ` should be deallocated at the end of the scope and asynchronous blocks below should not be called.